### PR TITLE
Fix timedistances

### PR DESCRIPTION
**Pull request by danich1**
This PR follows up on #7. You have already seen these results before so should be quick review.

Main bug fixes/enhancements are as follows:
1. The word vectors are actually aligned. Was aligning the word vectors but then resetting the alignment on accident.
2. The global and local distance plots make more sense as each year is compared to the a words origin year
3. Switched the low dimension model from TSNE to UMAP for consistency in later experiments.
4. I also changed how I'm calculating/representing word vectors to make coding easier.

In terms of review feel free to only look at notebooks with 03 prefix (those are the notebooks that contain the results).

### CODE REVIEW

File: biovectors_modules/word2vec_analysis_helper.py [hunk 1]
```diff
@@ -34,33 +34,70 @@ def generate_timeline(
...
-    """
-    This function is designed to get the global distance of ever token between two years.
-    Global distance is basically the cosine similarity (token_year_one, token_year_two)
+def _calculate_distances(
+    year_one_model: pd.DataFrame,
+    year_two_model: pd.DataFrame,
+    neighbors: int,
+    input_queue: JoinableQueue,
+    output_queue: JoinableQueue,
+):
```
Comment (miltondp) [inline biovectors_modules/word2vec_analysis_helper.py:43]: Is this function replacing both the previous global and local functions? Maybe adding docstring here might help.

File: biovectors_modules/word2vec_analysis_helper.py [hunk 2]
```diff
@@ -34,33 +34,70 @@ def generate_timeline(
...
+            ]
+        )
+
+        # Grab the similarity of word vectors
+        output_queue.put(
+            {
+                "token": tok,
+                "local_dist": pdist(
+                    1 - cdist(word_vec, neighbor_vec, "cosine"), "cosine"
+                ).item(),
```
Comment (miltondp) [inline biovectors_modules/word2vec_analysis_helper.py:96]: This is a bit hard to follow. If I understand correctly, you calculate the similarities between those two sets (`word_vec` and `neighbor_vec`), that is a word embedding representation (I think) and its neighbors, leading to an array (n x m). And then you calculate the distance (1 - sim) between those similarities (leading to n x n)? And this is the local distance, interesting! If that's correct, it was not hard to follow! :joy:

File: biovectors_modules/word2vec_analysis_helper.py [hunk 3]
```diff
@@ -75,53 +112,49 @@ def get_local_distance(
         shared_tokens - the list of all tokens shared across the years
         neighbors - the number of neighbor tokens to use
     """
+    token_distance = []
+    counter = 0
+    with Manager() as m:
+        tok_queue = m.JoinableQueue()
+        dist_queue = m.JoinableQueue()
```
Comment (miltondp) [inline biovectors_modules/word2vec_analysis_helper.py:131]: This looks very cool. I revisit this code the next time I need to run things in parallel.

File: biovectors_modules/word2vec_analysis_helper.py [hunk 4]
```diff
@@ -141,26 +173,34 @@ def get_neighbors(
...
-        token_neighbors = sim_mat[word_vector_row, :].argsort()[-neighbors:][::-1]
+        word_vector_matrix = word_vector_matrix.sort_values("token")
+
+        sim_mat = 1 - cdist(
+            word_vector_matrix.query(f"token == {repr(query_token)}")
+            .drop("token", axis=1)
+            .values,
+            word_vector_matrix.drop("token", axis=1).values,
+            "cosine",
+        )
```
Comment (miltondp) [inline biovectors_modules/word2vec_analysis_helper.py]: Minor: it might be better to factor out the two arrays you are trying to compare into two variables, and then pass them to `cdist`. It's a bit hard to read right now.

File: word2vec_decade_experiment/03_word_decade_figure_generator_2000_2020.py [hunk 5]
```diff
@@ -102,17 +110,17 @@
 local_distance
 
 g = plot_token_timeline(token_timeline_low_dim_df)
-g.save("output/figures/are_time_plot.png")
+g.save("output/timeline_figures/are_time_plot.png")
 print(g)
 
 plot_wordcloud_neighbors_gif(
-    token_timeline_low_dim_df, plot_filename="output/wordcloud_plots/are.gif"
+    token_timeline_low_dim_df, plot_filename="output/wordcloud_plots_2000_2020/are.gif"
```
Comment (miltondp) [inline word2vec_decade_experiment/03_word_decade_figure_generator_2000_2020.py]: Since you are reusing this folder, it might be better to put that in a variable.

File: word2vec_decade_experiment/03_word_decade_figure_generator_2000_2020.py [hunk 6]
```diff
@@ -294,22 +310,21 @@
...
 plot_wordcloud_neighbors(
-    token_timeline_low_dim_df, plot_filename="output/wordcloud_plots/cell.png"
+    token_timeline_low_dim_df, plot_filename="output/wordcloud_plots_2000_2020/cell.png"
 )
 
 # # Conclusions - Take Home Point(s)
 
-# 1. 2008 - 2010 has a huge spike in semantic change for all the tokens I have analyzed in this notebook
+# 1. Aligning vectors helps tremendously when measuring how words change their semantic meaning. Interesting spikes in the distance is words such as cystic, interleukin-18 and like 95%ci.
 # 2. Plotting word clouds for the neighbor of each word highlights the shift these vectors are capturing
```
Comment (miltondp) [inline word2vec_decade_experiment/03_word_decade_figure_generator_2000_2020.py:332]: Yes, looks great.

**Review by miltondp — APPROVED** [review]
Looks good to me! I left very minor comments that you can take or not.

File: biovectors_modules/word2vec_analysis_helper.py (hunk 2 above)
Comment (danich1) [reply biovectors_modules/word2vec_analysis_helper.py:96]: Yup! You got it. The local distance is a bit wonky of a description but basically its the distance of a tokens; similarity to all neighbors within both time periods.

File: biovectors_modules/word2vec_analysis_helper.py (hunk 1 above)
Comment (danich1) [reply biovectors_modules/word2vec_analysis_helper.py:43]: Yeah forgot to add that. Will do in next commit
